index.js

```diff
@@ -199,36 +200,25 @@
 
   /* istanbul ignore else */
   if (typeof module === 'object' && typeof module.exports === 'object') {
-    module.exports = f(require('ramda'), require('sanctuary-def'));
+    module.exports = f(require('sanctuary-type-classes'),
+                       require('sanctuary-def'));
   } else if (typeof define === 'function' && define.amd != null) {
-    define(['ramda', 'sanctuary-def'], f);
+    define(['sanctuary-type-classes', 'sanctuary-def'], f);
   } else {
-    self.sanctuary = f(self.R, self.sanctuaryDef);
+    self.sanctuary = f(self.sanctuaryTypeClasses, self.sanctuaryDef);
   }
 
-}(function(R, $) {
+}(function(Z, $) {
 
   'use strict';
 
-  var _slice = Array.prototype.slice;
   var _toString = Object.prototype.toString;
 
-  //  filter :: (Monad m, Monoid (m a)) => (a -> Boolean, m a) -> m a
-  function filter(pred, m) {
-    return m.chain(function(x) {
-      return pred(x) ? m.of(x) : m.empty();
-    });
-  }
-
-  //  hasMethod :: String -> a -> Boolean
-  function hasMethod(name) {
-    return function(x) {
-      return x != null && typeof x[name] === 'function';
-    };
-  }
+  //  Fn :: (Type, Type) -> Type
+  function Fn(x, y) { return $.Function([x, y]); }
 
-  //  negativeZero :: a -> Boolean
-  var negativeZero = R.either(R.equals(-0), R.equals(new Number(-0)));
+  //  Thunk :: Type -> Type
+  function Thunk(x) { return $.Function([x]); }
```

Comment:
@safareli, we can replace `$.Function([x])` with `$.Thunk(x)` once sanctuary-def supports this. :)